routes/habitRoutes.js

```diff
@@ -75,4 +75,22 @@ router.delete('/:id', passport.authenticate('jwt', { session: false }), (req, re
 		.catch(() => res.status(404).json({ message: 'Habit not found' }));
 });
 
+/**
+ * @route   PUT api/habit/:id
+ * @desc    Updates habit
+ * @access  Private
+ */
+router.put('/:id', passport.authenticate('jwt', { session: false }), (req, res) => {
+	const { name, description, type, difficulty, tags, frequency, user } = req.body;
+
+	// Check whether habit belongs to current authenticated user
+	Habit.findByIdAndUpdate(
+		{ _id: req.params.id },
+		{ $set: { name, description, type, difficulty, tags, frequency, user } },
```

Comment:
I don't think that we want to update user, once Habit is created its author should be set in stone.